src/onegov/org/models/document_form.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+from uuid import uuid4
+
+from onegov.chat.models.message import associated
+from onegov.core.collection import GenericCollection
+from onegov.core.orm import Base
+from onegov.core.orm.mixins import (
+    ContentMixin, TimestampMixin, content_property, dict_property)
+from onegov.core.orm.mixins.content import dict_markup_property
+from onegov.core.orm.types import UUID
+from onegov.core.utils import normalize_for_url
+from onegov.file import AssociatedFiles, File, SearchableFile
+from onegov.form import FormCollection
+from onegov.org.observer import observes
+from onegov.reservation import ResourceCollection
+from onegov.org.models import AccessExtension
+from onegov.search import SearchableContent
+from sqlalchemy import Column, Text
+
+
+from typing import Any, TYPE_CHECKING
+if TYPE_CHECKING:
+    import uuid
+    from sqlalchemy.orm import Query, Session
+
+
+class DocumentFormFile(File, SearchableFile):
```

Comment:
It might be better to include the file contents in the search data for `FormDocument` so you don't end up with separate search results for the PDF and the Form.

Add something like:
```python
    pdf_extract: dict_property[str | None] = content_property()

    @observes('pdf')
    def pdf_observer(self, pdf: DocumentFormFile | None):
        self.pdf_extract = pdf.extract if pdf is not None else None
```

to `FormDocument` and add `pdf_extract` to the `es_properties`.

You could probably also get away without duplicating the extract from the file to the document, but it would be slower during indexing, so it's probably worth duplicating. Just add a test to make sure the extract remains synced if the file gets replaced.